### PR TITLE
Zel

### PR DESCRIPTION
**Pull request by BogdanK0451**

### CODE REVIEW

File: WasteWar/Assets/Scripts/Camera Controls/Move.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
+﻿using UnityEngine;
+
+public class Move : MonoBehaviour
```
Comment (orenkoren) [inline WasteWar/Assets/Scripts/Camera Controls/Move.cs]: first unity functions like Start and Update, then your functions

File: WasteWar/Assets/Scripts/Camera Controls/Move.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    private Terrain terrain;
+    [SerializeField]
+    private Transform cam;
+
+    private Vector3 terrainSize;
+
+    void TryToMoveForward()
+    {
+        if (cam.position.z < (terrainSize.z - CameraConstants.WORLD_BORDER)
+           && (Input.GetKey("w") || Input.mousePosition.y >= Screen.height - CameraConstants.SCREEN_BORDER))
```
Comment (orenkoren) [inline WasteWar/Assets/Scripts/Camera Controls/Move.cs]: Use GetKey(KeyCode.W)

File: WasteWar/Assets/Scripts/Camera Controls/Move.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+public class Move : MonoBehaviour
+{
+    [SerializeField]
+    private Terrain terrain;
+    [SerializeField]
+    private Transform cam;
+
+    private Vector3 terrainSize;
+
+    void TryToMoveForward()
```
Comment (orenkoren) [inline WasteWar/Assets/Scripts/Camera Controls/Move.cs]: you're duplicated code with CameraConstants.SCREEN_BORDER
void TryToMoveForward()
{
if(keyboardpress || MouseAtBorder())
MoveForward()
}

File: WasteWar/Assets/Scripts/Camera Controls/Zoom.cs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    [SerializeField]
+    private Transform cam;
+    [SerializeField]
+    private Transform rotationRoute;
+
+    //parameter for parametric equation
+    public float t = 0;
+
+    //points along Bezier curve for smooth movement of camera
+    private Vector3 p1;
```
Comment (orenkoren) [inline WasteWar/Assets/Scripts/Camera Controls/Zoom.cs:16]: meaningful names please

File: WasteWar/Assets/Scripts/Camera Controls/Zoom.cs [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    {
+        p1 = route.GetChild(0).position;
+        p2 = route.GetChild(1).position;
+        p3 = route.GetChild(2).position;
+        p4 = route.GetChild(3).position;
+    }
+
+    private void SetCameraZoomLevel(bool initial,bool increment)
+    {
+        if(initial)
```
Comment (orenkoren) [inline WasteWar/Assets/Scripts/Camera Controls/Zoom.cs]: rename those bools... bools start with should or can

File: WasteWar/Assets/Scripts/Camera Controls/Zoom.cs [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        while (cam.rotation.eulerAngles.x > 60)
```
Comment (orenkoren) [inline WasteWar/Assets/Scripts/Camera Controls/Zoom.cs]: why 60? const that into a logical name

File: WasteWar/Assets/Scripts/Camera Controls/Zoom.cs [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+        {
+            IncreaseParameter(increment);
```
Comment (orenkoren) [inline WasteWar/Assets/Scripts/Camera Controls/Zoom.cs]: i dont undertsand what this parameter is

Comment (orenkoren) [reply WasteWar/Assets/Scripts/Camera Controls/Zoom.cs]: also those 3 lines are duplicated.. pull into function

File: WasteWar/Assets/Scripts/Camera Controls/Zoom.cs [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+    {
+        return Quaternion.Lerp(
+                   rotationRoute.GetChild(0).rotation,
+                   rotationRoute.GetChild(1).rotation,
+                   t);
+    }
+    //using the parameter and the given curve to specify where the gameObject will be positioned
+    private Vector3 CalculateCurrentPosBasedOnParameterT()
+    {
+        return Mathf.Pow(1 - t, 3) * p1 +
```
Comment (orenkoren) [inline WasteWar/Assets/Scripts/Camera Controls/Zoom.cs]: need an explanation for this either in the variable names or document this function

File: WasteWar/Assets/Scripts/Camera Controls/ZoomRouteDraw.cs [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+{
+    [SerializeField]
+    private Transform[] controlPoints;
+    private Vector3 gizmosPosition;
+
+    private void OnDrawGizmos()
+    {
+        for (float t=0; t <= 1; t += 0.02f)
+        {
+            gizmosPosition = Mathf.Pow(1 - t, 3) * controlPoints[0].position +
```
Comment (orenkoren) [inline WasteWar/Assets/Scripts/Camera Controls/ZoomRouteDraw.cs]: is this the same as the function from Zoom.cs? if so pull it to MathUtils.cs or smth and access that from both

File: WasteWar/Assets/Scripts/Camera Controls/Zoom.cs (hunk 4 above)
Comment (BogdanK0451) [reply WasteWar/Assets/Scripts/Camera Controls/Zoom.cs:16]: gotta disagree on this one, having long variable names in math equations can lead to it being harder understand than having short names + comment above, if this was general purpose code for someone that knows 0 math then sure, but in our scenario i believe it's unnecessary renaming this one when there's a comment explaining what it is, or a rename could be done but the name has to be very short, which i can't think of right now

**Review by orenkoren — APPROVED** [review]

**Pull request merged**